packages/menu/src/MenuItem.tsx

```diff
@@ -63,6 +65,13 @@ interface BaseMenuItemProps {
   children?: React.ReactNode;
 
   href?: string;
+
+  /**
+   * Determines whether or not the component will be rendered in dark mode.
+   *
+   * default: `false`
+   */
+  darkMode?: boolean;
```

Comment:
I think we should use an internal `MenuContext` for this, or use an `InternalMenuItem` component like we do with `Select`. It feels confusing to have a consumer-facing prop like this that doesn't do anything

Reply:
I see, thats similar to what the select component. Yeah that makes more sense.

Reply:
We also have an extendable `DarkModeProps` interface now! For all places in this PR that define `darkMode` in interfaces